packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -1389,7 +1389,14 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
             }
         }
 
-        this.stats.errorTracker.add(error);
+        // If the request is non-retryable, the error and snapshot aren't saved in the errorTrackerRetry object.
+        // Therefore, we pass the crawlingContext to the errorTracker.add method, enabling snapshot capture.
+        // This is to make sure the error snapshot is not duplicated in the errorTrackerRetry and errorTracker objects.
```

Comment:
This is unrelated to this PR but I don't see much use in having separate retry and non-retry error trackers, I would be for merging them cc @B4nan

Reply:
In the [SDK_CRAWLER_STATISTICS](https://api.apify.com/v2/key-value-stores/L7DclFFX3fHuPCne9/records/SDK_CRAWLER_STATISTICS_0) file we have: `retryErrors` and `errors`, and in [basic-crawler.ts](https://github.com/apify/crawlee/pull/2332/files/5f50e30181dfd3e680e2d82c373bdd5e1e619d06#diff-3606a49356303bfa2bfed957bc81701e3d45d4cdafd4d5182685ef826c93b63f) file we have `errorTrackerRetry` and `errorTracker`.

`errorTrackerRetry` only works if the request retries, and `errorTracker` works when there are no retries or when we reach the maximum retries.

Reply:
Yeah I know, I just don't know the reason to have it split like this, for me it has no value if it was the last error or not.

Reply:
What do you think @B4nan @vladfrangu? Do you see any advantage in keeping those separate? Merging them would simplify the code and the statistics object and analysis. Not sure if we would consider changing this file structure a breaking change.